pkg/featuregate/featuregate.go

```diff
@@ -56,6 +56,12 @@ var (
 		"operator.collector.rewritetargetallocator",
 		featuregate.StageAlpha,
 		featuregate.WithRegisterDescription("controls whether the operator should configure the collector's targetAllocator configuration"))
+
+	// PrometheusOperatorIsAvailable is the feature gate that enables features associated to the Prometheus Operator.
+	PrometheusOperatorIsAvailable = featuregate.GlobalRegistry().MustRegister(
+		"operator.observability.prometheus",
+		featuregate.StageAlpha,
+		featuregate.WithRegisterDescription("enables features associated to the Prometheus Operator"))
```

Comment:
Please add a `featuregate.FromVersion` that marks the Operator version this feature gate was added. (This was a rather late concept in the featuregate package and we should go back and fix our other gates to include it)